src/library/scala/collection/mutable/TreeMap.scala

```diff
@@ -0,0 +1,163 @@
+package scala
+package collection
+package mutable
+
+import scala.collection.generic._
+import scala.collection.mutable.{RedBlackTree => RB}
+
+/**
+ * $factoryInfo
+ *
+ * @define Coll mutable.TreeMap
+ * @define coll mutable tree map
+ */
+object TreeMap extends MutableSortedMapFactory[TreeMap] {
+
+  def empty[A, B](implicit ord: Ordering[A]) = new TreeMap[A, B]()(ord)
+
+  /** $sortedMapCanBuildFromInfo */
+  implicit def canBuildFrom[A, B](implicit ord: Ordering[A]): CanBuildFrom[Coll, (A, B), TreeMap[A, B]] =
+    new SortedMapCanBuildFrom[A, B]
+}
+
+/**
+ * A mutable sorted map implemented using a mutable red-black tree as underlying data structure.
+ *
+ * @param ordering the implicit ordering used to compare objects of type `A`.
+ * @tparam A the type of the keys contained in this tree map.
+ * @tparam B the type of the values associated with the keys.
+ *
+ * @author Rui Gonçalves
+ * @version 2.12
+ * @since 2.12
+ *
+ * @define Coll mutable.TreeMap
+ * @define coll mutable tree map
+ */
+class TreeMap[A, B] private (tree: RB.Tree[A, B])(implicit val ordering: Ordering[A])
```

Comment:
In this case I can't do that because `TreeMapView` extends `TreeMap`. This is needed because `rangeImpl` in `SortedMapLike` requires implementations to return the same class of the original (the `This` type argument).

When `rangeImpl` is called, the immutable version creates a new `immutable.TreeMap` with the same class of the original, and `mutable.TreeSet` has `from` and `until` bounds in the original class already. I didn't follow the example of `mutable.TreeSet` because that would require always checking the defined bounds on each call to tree map operations, regardless of whether it is a range view or the whole collection. As I think the most common use case for a tree map is to do operations on it without acting on views, I decided to separate the two "code bases".